caom2persistence/src/main/java/ca/nrc/cadc/caom2/persistence/BaseSQLGenerator.java

```diff
@@ -751,7 +750,7 @@ public String getSelectArtifactSQL(URI artifactURI) {
         sb.append(getTable(Artifact.class));
         sb.append(" WHERE ");
         sb.append("uri='");
-        sb.append(artifactURI.toString());
+        sb.append(artifactKey.toString());
```

Comment:
Although executed via a PreparedStatementCreator, the uri value is being stuck into the string rather than bound... SQL injection.